src/modules/userCreationRequestMethods.ts

```diff
@@ -570,8 +574,81 @@ export const userCreationRequestsMethods = {
 
         updateData.organization = { connect: { id: organizationUnitId } };
 
+        const createNewSupplementalPositions = [];
+
+        const mainPosition = requestBeforeUpdate.supplementalPositions.find((position) => position.main);
+
+        if (!mainPosition) {
+            const newMainPosition = {
+                organizationUnit: { connect: { id: editData.organizationUnitId } },
+                percentage: (editData.percentage || 1) * percentageMultiply,
+                main: true,
+                role: editData.title,
+                status: PositionStatus.ACTIVE,
+                workStartDate: editData.date,
+            };
+            createNewSupplementalPositions.push(newMainPosition);
+        }
+
+        if (
+            mainPosition &&
+            (editData.title !== mainPosition.role ||
+                mainPosition.organizationUnitId !== editData.organizationUnitId ||
+                editData.date !== mainPosition.workStartDate ||
+                editData.unitId !== mainPosition.unitId ||
+                (editData.percentage && editData.percentage * percentageMultiply !== mainPosition.percentage))
+        ) {
+            await prisma.supplementalPosition.update({
+                where: { id: mainPosition.id },
+                data: {
+                    role: editData.title,
+                    workStartDate: editData.date,
+                    unitId: editData.unitId || null,
+                    percentage: (editData.percentage || 1) * percentageMultiply,
+                    organizationUnitId: editData.organizationUnitId,
+                },
+            });
+        }
+
+        const supplementalPositionBefore = requestBeforeUpdate.supplementalPositions.find((position) => !position.main);
+
+        if (supplementalPositions && supplementalPositions[0].organizationUnitId && !supplementalPositionBefore) {
+            const newSupplementalPosition = {
+                organizationUnit: { connect: { id: supplementalPositions[0].organizationUnitId } },
+                percentage: (supplementalPositions[0].percentage || 1) * percentageMultiply,
+                main: false,
+                role: editData.title,
+                status: PositionStatus.ACTIVE,
+                workStartDate: editData.date,
+                unitId: supplementalPositions[0].unitId,
+            };
+            createNewSupplementalPositions.push(newSupplementalPosition);
+        }
+
+        if (
+            supplementalPositionBefore &&
+            supplementalPositions &&
+            supplementalPositions[0] &&
+            (editData.title !== supplementalPositionBefore.role ||
+                editData.date !== supplementalPositionBefore.workStartDate ||
+                supplementalPositions[0].organizationUnitId !== supplementalPositionBefore.organizationUnitId ||
+                supplementalPositions[0].unitId !== supplementalPositionBefore.unitId ||
```

Comment:
if you put out the supplementalPositions[0] into a variable and guard return it it would make this code 10x shorter probably